lib/types/inet-address.js

```diff
@@ -10,238 +11,269 @@ const utils = require("../utils");
-    if (!value) {
-        return new InetAddress(utils.allocBufferFromArray([0, 0, 0, 0]));
-    }
-    const ipv4Pattern = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/;
-    const ipv6Pattern = /^[\da-f:.]+$/i;
-    let parts;
-    if (ipv4Pattern.test(value)) {
-        parts = value.split(".");
-        return new InetAddress(utils.allocBufferFromArray(parts));
+    get version() {
+        return this.#internal.getVersion();
     }
-    if (!ipv6Pattern.test(value)) {
-        throw new TypeError(
-            "Value could not be parsed as InetAddress: " + value,
-        );
-    }
-    parts = value.split(":");
-    if (parts.length < 3) {
-        throw new TypeError(
-            "Value could not be parsed as InetAddress: " + value,
-        );
+
+    set version(value) {
+        throw new SyntaxError("InetAddress version is read-only");
     }
-    const buffer = utils.allocBufferUnsafe(16);
-    let filling = 8 - parts.length + 1;
-    let applied = false;
-    let offset = 0;
-    const embeddedIp4 = ipv4Pattern.test(parts[parts.length - 1]);
-    if (embeddedIp4) {
-        // Its IPv6 address with an embedded IPv4 address:
-        // subtract 1 from the potential empty filling as ip4 contains 4 bytes instead of 2 of a ipv6 section
-        filling -= 1;
+
+    /**
+     * Immutable buffer that represents the IP address
+     * @readonly
+     * @type Array
+     */
+    get buffer() {
+        return this.#internal.getBuffer();
     }
-    function writeItem(uIntValue) {
-        buffer.writeUInt8(+uIntValue, offset++);
+
+    set buffer(value) {
+        throw new SyntaxError("InetAddress buffer is read-only");
     }
-    for (let i = 0; i < parts.length; i++) {
-        const item = parts[i];
-        if (item) {
-            if (embeddedIp4 && i === parts.length - 1) {
-                item.split(".").forEach(writeItem);
-                break;
-            }
-            buffer.writeUInt16BE(parseInt(item, 16), offset);
-            offset = offset + 2;
-            continue;
+
+    /**
+     * Converts a string representation of an IP address to an InetAddress instance.
+     *
+     * This function accepts both IPv4 and IPv6 addresses. For IPv4 addresses, it expects
+     * a string in the format "xxx.xxx.xxx.xxx" where each "xxx" is a number between 0 and 255.
+     * For IPv6 addresses, it expects a string in the standard IPv6 format, which may include
+     * an embedded IPv4 address.
+     *
+     * @param {string} value - The string representation of the IP address.
+     * @returns {InetAddress} - An instance of InetAddress representing the given IP address.
+     * @throws {TypeError} - If the input string is not a valid IPv4 or IPv6 address.
+     */
+    static fromString(value) {
+        if (!value) {
+            return new InetAddress(utils.allocBufferFromArray([0, 0, 0, 0]));
+        }
+        const ipv4Pattern = /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$/;
+        const ipv6Pattern = /^[\da-f:.]+$/i;
```

Comment:
Provide more description of this regexes. Maybe you can use a more complcated one but skip some of the checks?

Reply:
I'll add reference to the source of this regex